### DISCUSSION TITLE
Allow field to be populated from multiple aliases

### DISCUSSION THREAD

**Discussion by muety** [pydantic/pydantic#3988] [Feature Request]
When de-serializing some JSON payload to a Pydantic model, I'd like to accept multiple source names for a given field. I came across the `alias` keyword, but it only accepts a single string, rather than a list and also affects serialization in addition. 

## Example
```python
class Foo(BaseModel):
  bar: str = Field(aliases=['Bar', 'baz'])
```

In the example, id like all of the following three inputs be valid for instantiating a `Foo` object:
* `{ "bar": "..." }`
* `{ "Bar": "..." }`
* `{ "baz": "..." }`

Also, is there a way to generally be case-insensitive during de-serialization, i.e. always accept `Bar` for populating field `bar`?

**Comment by Rawk** [comment]
For being case-insensitive, i think you can use [alias-generator](https://pydantic-docs.helpmanual.io/usage/model_config/#alias-generator).
